account-kit/smart-contracts/src/ma-v2/client/client.test.ts

```diff
@@ -182,6 +191,64 @@ describe("MA v2 Tests", async () => {
     ).rejects.toThrowError();
   });
 
+  it("successfully sign + validate a message, for native and single signer validation", async () => {
+    const provider = (await givenConnectedProvider({ signer })).extend(
+      installValidationActions
+    );
+
+    const accountContract = getContract({
+      address: provider.getAddress(),
+      abi: semiModularAccountBytecodeAbi,
+      client,
+    });
+
+    // UO deploys the account to test 1271 against
+    const result = await provider.installValidation({
+      validationConfig: {
+        moduleAddress: getDefaultSingleSignerValidationModuleAddress(
+          provider.chain
+        ),
+        entityId: 1,
+        isGlobal: true,
+        isSignatureValidation: true,
+        isUserOpValidation: true,
+      },
+      selectors: [],
+      installData: SingleSignerValidationModule.encodeOnInstallData({
+        entityId: 1,
+        signer: await sessionKey.getAddress(),
+      }),
+      hooks: [],
+    });
+
+    let txnHash = provider.waitForUserOperationTransaction(result);
+    await expect(txnHash).resolves.not.toThrowError();
```

Comment:
As mentioned in other PRs, `resolves.not.toThrowError()` isn't particularly better than just awaiting the promise, e.g.
```ts
await provider.waitForUserOperationTransaction(result);
```